src/scripts/deactivate-user.ts

```diff
@@ -14,14 +16,31 @@ const deactivateUser = async (
 ): Promise<User> => {
   const client = oktaManageClient(oktaConfiguration);
 
-  const oktaUser = await client.getUser(userId);
+  const maybeOktaUser = await getUser(userId, client);
 
-  // eslint-disable-next-line functional/no-expression-statement
-  await oktaUser.deactivate();
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  const deactivate = async (oktaUser: OktaUser) => {
+    // eslint-disable-next-line functional/no-expression-statement
+    await oktaUser.deactivate({
+      sendEmail: false,
+    });
+
+    const deactivatedOktaUser = await client.getUser(userId);
+
+    return oktaUserAsUser(deactivatedOktaUser);
+  };
 
-  const decativatedOktaUser = await client.getUser(userId);
+  // eslint-disable-next-line functional/functional-parameters
+  const throwOnMissing = () => {
+    // eslint-disable-next-line functional/no-throw-statement
+    throw new Error(`User [${userId}] does not exist. Can not de-activate.`);
+  };
 
-  return user(decativatedOktaUser);
+  return maybeOktaUser === undefined
+    ? throwOnMissing()
+    : maybeOktaUser.status === UserStatus.DEPROVISIONED
+    ? oktaUserAsUser(maybeOktaUser)
```

Comment:
Add a warning that the user has already been de-activated to avoid any confusion

Reply:
My *weak* preference is that a no-op succeeds. I'm happy to make this a warning or error though.